### PR TITLE
Add audio preview to Inspector dock

### LINKED ISSUE #20181: No way to just preview sounds
Godot 3.0.5 and master 085483e88577ea91400a1ee46bc4321340824c51

I started importing sounds for the first time in my project, and I rapidly found annoying it's not possible to listen to them. When I double-click on a sound in the file explorer, all I see is this:

image

### PR DESCRIPTION
**Pull request by volzhs**
can play / pause / stop / seek audio.
close #20181

sound_preview

### CODE REVIEW

**Comment by fracteed** [comment]
Awesome, this will be very useful! Is there any chance of also adding a play button to the AudioSteramPlayers themselves. I find that toggling the "playing" checkbox on and off often doesn't work, and it is very usefull to be able to check levels, buss fx and randomisation directly.

**Comment by volzhs** [comment]
@fracteed I not sure about adding a play button to AudioSteramPlayer.
open an issue for it. :)

**Comment by akien-mga** [comment]
Great feature! I'm just unsure about the `AudioEditor` class name and file name, it seems a bit broad for what is an audio stream preview plugin and not an audio editor (you can't edit audio tracks with it).

Currently preview plugins are in `editor/plugins/editor_preview_plugins.cpp` with the name `Editor<Resource>PreviewPlugin`, so here it could be put there and named `EditorAudioStreamPreviewPlugin`.

There's also `editor/audio_stream_preview.cpp` with `AudioStreamPreview` and `AudioStreamPreviewGenerator`. Sounds like there's some potential for unifying all that.

**Comment by volzhs** [comment]
yeah. I have a doubt about th name of class too. it was originally copied from TextureEditor `editor/plugins/texture_editor_plugin.h, cpp`.
I will update it as you suggested.

**Comment by reduz** [comment]
agreed about the naming, should probably be AudioStreamEditor, given the resource type is AudioStream

File: editor/plugins/audio_stream_editor_plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,284 @@
+/*************************************************************************/
+/*  audio_editor_plugin.cpp                                              */
```
Comment (allkhor) [inline editor/plugins/audio_stream_editor_plugin.cpp]: Mismatched comment and the file name, also in .h file.

Comment (volzhs) [reply editor/plugins/audio_stream_editor_plugin.cpp]: ouch...

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**